pkg/util/kubernetes/hpa/hpa.go

```diff
@@ -66,3 +66,12 @@ func DiffExternalMetrics(informerList []*autoscalingv2.HorizontalPodAutoscaler,
 	}
 	return
 }
+
+// AutoscalerMetricsUpdate will return true if the applied configuration of the Autoscaler has changed.
+// We only care about updates of the metrics or their scopes.
+func AutoscalerMetricsUpdate(new, old *autoscalingv2.HorizontalPodAutoscaler) bool {
+	if reflect.DeepEqual(new.Annotations, old.Annotations) {
```

Comment:
return !reflect.DeepEqual(new.Annotations, old.Annotations)

Reply:
What is in this `Annotations` field? If it's the HPA object's annotations that won't work, they get updated frequently with fields like `lastTransitionTime`, so we will always see an update?